src/pop/blockchain/alt_block_tree.cpp

```diff
@@ -409,10 +531,20 @@ bool AltBlockTree::loadBlock(const stored_index_t& index,
   if (!checkIdsForDuplicates<VTB>(vtbIds, state)) return false;
   if (!checkIdsForDuplicates<ATV>(atvIds, state)) return false;
 
-  if (hasDuplicates<VbkBlock>(*current, vbkblockIds, *this, state) ||
-      hasDuplicates<VTB>(*current, vtbIds, *this, state) ||
-      hasDuplicates<ATV>(*current, atvIds, *this, state)) {
-    return false;
+  if (current->pprev != nullptr) {
+    // if the block is not yet connected, defer the stateful duplicate check to
+    // connectBlock()
+    if (current->isConnected()) {
```

Comment:
As we have the same validation inside the connectBlock() why we can not remove validation from this place ?

Reply:
As far as I understand this is just a sanity check that is supposed to verify that connectBlock did its job. I think I see some other issues with loadBlock, but that belongs to another patch. I fixed this specific issue since it was surfaced by this patch.